android-components/components/feature/downloads/src/test/java/mozilla/components/feature/downloads/temporary/CopyDownloadFeatureTest.kt

```diff
@@ -0,0 +1,344 @@
+
+package mozilla.components.feature.downloads.temporary
+
+import android.content.Context
+import android.view.View
+import android.webkit.MimeTypeMap
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import kotlinx.coroutines.test.runTest
+import mozilla.components.browser.state.action.CopyInternetResourceAction
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.browser.state.state.ContentState
+import mozilla.components.browser.state.state.TabSessionState
+import mozilla.components.browser.state.state.content.ShareInternetResourceState
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.fetch.Client
+import mozilla.components.concept.fetch.Headers.Names.CONTENT_TYPE
+import mozilla.components.concept.fetch.MutableHeaders
+import mozilla.components.concept.fetch.Request
+import mozilla.components.concept.fetch.Response
+import mozilla.components.support.test.any
+import mozilla.components.support.test.argumentCaptor
+import mozilla.components.support.test.ext.joinBlocking
+import mozilla.components.support.test.mock
+import mozilla.components.support.test.robolectric.testContext
+import mozilla.components.support.test.rule.MainCoroutineRule
+import mozilla.components.support.utils.SnackbarDelegate
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.doAnswer
+import org.mockito.Mockito.doNothing
+import org.mockito.Mockito.doReturn
+import org.mockito.Mockito.spy
+import org.mockito.Mockito.verify
+import org.robolectric.Shadows.shadowOf
+import java.io.File
+import java.nio.charset.StandardCharsets
+
+/**
+ * The 89a gif header as seen on https://www.w3.org/Graphics/GIF/spec-gif89a.txt
+ */
+private const val GIF_HEADER = "GIF89a"
+
+@RunWith(AndroidJUnit4::class)
+class CopyDownloadFeatureTest {
+    // When writing new tests initialize CopyDownloadFeature with this class' context property
+    // When creating new directories use class' context property#cacheDir as a parent
+    // This will ensure the effectiveness of @After. Otherwise leftover files may be left on the machine running tests.
+
+    private lateinit var context: Context
+    private val testCacheDirName = "testCacheDir"
+
+    @get:Rule
+    val coroutinesTestRule = MainCoroutineRule()
+    private val dispatcher = coroutinesTestRule.testDispatcher
+    private val scope = coroutinesTestRule.scope
+
+    @Before
+    fun setup() {
+        // Effectively reset context mock
+        context = spy(testContext)
+        doReturn(File(testCacheDirName)).`when`(context).cacheDir
+    }
+
+    @After
+    fun cleanup() {
+        context.cacheDir.deleteRecursively()
+    }
+
+    @Test
+    fun `cleanupCache should automatically be called when this class is initialized`() = runTest {
```

Comment:
nit: new tests should follow the `GIVEN-WHEN-THEN` naming scheme.

Reply:
👍 My thoughts on this were as this is derived from the original ShareDownloadFeatureTest it would be easier to review and maintain if they were consistent in structure and naming?